### ISSUE TITLE
No coverage of --change-set-name nor parallel deployments with `cdk deploy`

### ISSUE THREAD

**Issue by shellscape** [awsdocs/aws-cdk-guide#330]
As requested by the issue template for https://github.com/aws/aws-cdk/issues/14344, I'm opening a separate issue here to track the lack of documentation in the CDK developer guide (https://docs.aws.amazon.com/cdk/latest/guide/cli.html) around parallel deployments with `cdk deploy`, known issues around that, and the remedy to a particular issue that arises. I would like to suggest adding documentation to the linked page and would be happy to submit a PR for that. Please let me know if that would be accepted.

**Comment by shellscape** [comment]
@Jerry-AWS I have significant experience in authoring docs for open source, and would be happy to contribute this change.

**Comment by Jerry-AWS** [comment]
@shellscape  Please feel free to give it a whack. Contributions are always appreciated!

**Comment by shellscape** [comment]
@Jerry-AWS will do. I think it'll be prudent to wait for the outcome of https://github.com/aws/aws-cdk/issues/14350, as CDK may not support any kind of concurrency. Please do keep this open pending resolution of that issue.

**Comment by shellscape** [comment]
I'm not sure that @rix0rrr understood fully what I was reporting there. CloudFront definitely doesn't support concurrent deployments of the _same_ stack, but it does support concurrent deployments of _different_ stacks, with which I think that CDK CLI still has a nasty bug. We could safely document the forbidden concurrent-same-stack-deployments, but should probably still wait on resolution of the bug reported.
